### DISCUSSION TITLE
Unable to Join Host

### DISCUSSION THREAD

**Discussion by pattreahfl24** [diangogav/EDOpro-server-ts#135] [Q&A] [answered]
Hello I'm trying to make a private server for my personal use, I brought a vps and domain name, I successfully installed the project to debian OS and run it but I can't join the host. please help.

![Screenshot 2024-09-24 010925](https://github.com/user-attachments/assets/d20a9fe9-b638-4d23-aaa1-28f9a2287432)

![Screenshot 2024-09-24 010852](https://github.com/user-attachments/assets/e97d2130-4f48-44e6-a58c-44eb9c6f2ba6)

**Comment by diangogav** [comment]
Hi, have you verified that you have ports 7711 and 7911 open? also for this screen you must use port 7711

**Reply by pattreahfl24** [reply]
Thanks @diangogav I didn't know about 7711 port, I have some progress but I'm still getting some errors 
![Screenshot 2024-09-24 055808](https://github.com/user-attachments/assets/475195b9-77d1-4531-862f-31f443237453)

**Reply by diangogav** [reply, marked as answer]
Are you using docker to build the project? if not, you must install all the dependency libraries.
`liblua5.3-dev` `libsqlite3-dev` `libevent-dev`
in the case of the screenshot you are sharing it says that `liblua5.3-dev` is missing

**Reply by pattreahfl24** [reply]
It's now working thanks!!! 

A few questions though..
1. We are under OCG region for now I'm using the OT as password but still the Forbidden list is set to 2024.09 TCG when I'm entering the room. How can I set it to default OCG banlist?
2. When I close the terminal and tried to enter a room It's disconnecting seems the server is closing, do the docker solves that?

**Reply by diangogav** [reply]
1. In the `MercuryBanListMemoryRepository` class, you need to implement the `getLastOCGIndex` method and use it in the `MercuryRoom` class on line 93. You can reference the `getLastTCGIndex` method.

2. You need to run Node.js as a background service: https://medium.com/@techclaw/running-a-node-js-app-as-a-background-service-a-step-by-step-guide-c1077c8aa2f7. If you are using Docker, you only need to run the Docker container in detached mode.

**Reply by pattreahfl24** [reply]
I'm not sure what went wrong..

the commands I used was 
docker build -t dockerfile /home/debian/EDOpro-server-ts
docker run -d -p 7911:7911 -p 7922:7922 dockerfile

![Screenshot 2024-09-24 231812](https://github.com/user-attachments/assets/8b7bf42d-0523-48ea-9c9d-06bb2ba8d871)

**Reply by diangogav** [reply]
`docker run -d -p 7911:7911 -p 7922:7922 -P 7711:7711 dockerfile`
Port 7711 again

**Reply by pattreahfl24** [reply]
It returns Unable to find image '7711:7711' locally

![Screenshot 2024-09-25 010331](https://github.com/user-attachments/assets/3348b2b0-78d8-49e8-96c0-81578f6e520d)

**Reply by pattreahfl24** [reply]
okay I notice the -P before 7711 needs to be lowercase

**Reply by diegofcornejo** [reply]
```sh
# Go to the cloned repository
cd /path/to/EDOpro-server-ts

# Build the docker image
docker build -t evolutionygo/server --no-cache --progress=plain .

# Run the docker image
docker run -d \
  --name evolutionygo-server \
  -p 7711:7711 \
  -p 7911:7911 \
  -p 7922:7922 \
  evolutionygo/server

```

**Reply by pattreahfl24** [reply]
How can I show or view the duel statistics via web? just like on evolution website?

**Comment by pattreahfl24** [comment]
I want to use docker but i dont know how to run it the command is there but
I dont know where is the image that I will be using

**Reply by diangogav** [reply]
You should install docker in your server and run the container in detached mode
